public/app/features/explore/Logs/Logs.tsx

```diff
@@ -156,81 +141,90 @@ const getDefaultVisualisationType = (): LogsVisualisationType => {
-    if (
-      this.props?.panelState?.logs?.columns ||
-      this.props?.panelState?.logs?.refId ||
-      this.props?.panelState?.logs?.labelFieldName
-    ) {
-      dispatch(
-        changePanelState(this.props.exploreId, 'logs', {
-          ...this.props.panelState?.logs,
-          columns: undefined,
-          visualisationType: this.state.visualisationType,
-          labelFieldName: undefined,
-          refId: undefined,
-        })
-      );
-    }
-  }
-
-  registerLogLevelsWithContentOutline = () => {
+const UnthemedLogs: React.FunctionComponent<Props> = (props: Props) => {
+  const {
+    width,
+    splitOpen,
+    logRows,
+    logsMeta,
+    logsVolumeEnabled,
+    logsVolumeData,
+    loadLogsVolumeData,
+    loading = false,
+    onClickFilterLabel,
+    onClickFilterOutLabel,
+    timeZone,
+    scanning,
+    scanRange,
+    showContextToggle,
+    absoluteRange,
+    onChangeTime,
+    getFieldLinks,
+    theme,
+    logsQueries,
+    clearCache,
+    addResultsToCache,
+    exploreId,
+    getRowContext,
+    getLogRowContextUi,
+    getRowContextQuery,
+    loadMoreLogs,
+    panelState,
+    eventBus,
+  } = props;
+  const [showLabels, setShowLabels] = useState<boolean>(store.getBool(SETTINGS_KEYS.showLabels, false));
+  const [showTime, setShowTime] = useState<boolean>(store.getBool(SETTINGS_KEYS.showTime, true));
+  const [wrapLogMessage, setWrapLogMessage] = useState<boolean>(store.getBool(SETTINGS_KEYS.wrapLogMessage, true));
+  const [prettifyLogMessage, setPrettifyLogMessage] = useState<boolean>(
+    store.getBool(SETTINGS_KEYS.prettifyLogMessage, false)
+  );
+  const [dedupStrategy, setDedupStrategy] = useState<LogsDedupStrategy>(LogsDedupStrategy.none);
+  const [hiddenLogLevels, setHiddenLogLevels] = useState<LogLevel[]>([]);
+  const [logsSortOrder, setLogsSortOrder] = useState<LogsSortOrder>(
+    store.get(SETTINGS_KEYS.logsSortOrder) || LogsSortOrder.Descending
+  );
+  const [isFlipping, setIsFlipping] = useState<boolean>(false);
+  const [displayedFields, setDisplayedFields] = useState<string[]>([]);
+  const [forceEscape, setForceEscape] = useState<boolean>(false);
+  const [contextOpen, setContextOpen] = useState<boolean>(false);
+  const [contextRow, setContextRow] = useState<LogRowModel | undefined>(undefined);
+  const [visualisationType, setVisualisationType] = useState<LogsVisualisationType | undefined>(
+    panelState?.logs?.visualisationType ?? getDefaultVisualisationType()
+  );
+  const [logsContainer, setLogsContainer] = useState<HTMLDivElement | undefined>(undefined);
+  const dispatch = useDispatch();
+  const previousLoading = usePrevious(loading);
+
+  const logsVolumeEventBus = eventBus.newScopedBus('logsvolume', { onlyLocal: false });
+  const { register, unregisterAllChildren } = useContentOutlineContext() ?? {};
+  let flipOrderTimer: number | undefined = undefined;
+  let cancelFlippingTimer: number | undefined = undefined;
```

Comment:
```suggestion
  let flipOrderTimer: number | undefined = undefined;
  let cancelFlippingTimer: number | undefined = undefined;
```

Shouldn't these be refs? I have no way of telling if `useUnmount` would receive the expected values when it executes. Worst case scenario, using refs would have no unwanted side effects.

Reply:
useUnmount uses refs under the hood - https://github.com/streamich/react-use/blob/master/src/useUnmount.ts

Reply:
Right, but on every render() call `flipOrderTimer` and `cancelFlippingTimer` will be created with undefined value, potentially overriding (or forgetting) the previous value. Before we used class attributes for that, which hold the same value as long as the component instance lives.

Reply:
Addressed in https://github.com/grafana/grafana/pull/87808/commits/55b82654ab0c6dbffef5a3164c486829a288a301